driver/compmisc.ml

```diff
@@ -19,6 +19,29 @@
    then the standard library directory (unless the -nostdlib option is given).
  *)
 
+let init_path ?(dir="") () =
+  let dirs =
+    if !Clflags.use_threads then "+threads" :: !Clflags.include_dirs
+    else
+      !Clflags.include_dirs
+  in
+  let dirs =
+    !Compenv.last_include_dirs @ dirs @ Config.flexdll_dirs @
+    !Compenv.first_include_dirs
+  in
+  let exp_dirs =
+    List.map (Misc.expand_directory Config.standard_library) dirs 
+  in
+  let tempname = 
+    if !Clflags.remove_cwd then 
+      List.rev_append exp_dirs (Clflags.std_include_dir ()) @ [dir]
+    else 
+      dir :: List.rev_append exp_dirs (Clflags.std_include_dir ()) 
+  in
+  Load_path.init(tempname);
+  Env.reset_cache ()
```

Comment:
Meta-comment: it is better to modify the code in-place, instead of duplicating it and modifying it, as this enlarges the diff artificially and makes it hard it review.